src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -2590,16 +2621,26 @@ public struct ParamMarshal
 
             paramMarshal.Context = ctx;
 
-            var marshal = new CSharpMarshalManagedToNativePrinter(ctx);
-            param.CSharpMarshalToNative(marshal);
+            if (param.IsInOut && param.Type.IsPointerToPrimitiveType()
+                && ExtensionMethods.AllowedToHaveDefaultPtrVals.Any(primType => param.Type.IsPointerToPrimitiveType(primType)))
```

Comment:
Since you always use this list the same way, why don't you just create a function that takes a primitive type and returns a Boolean?